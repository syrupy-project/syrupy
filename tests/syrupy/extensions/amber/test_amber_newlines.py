class ReprWithNewline:
    def __init__(self, newlines: int = 1) -> None:
        self.newlines = newlines

    def __repr__(self) -> str:
        newlines = "\n" * self.newlines
        return f"ReprWithNewline{newlines}"


def test_trailing_no_newline_in_repr(snapshot):
    assert ReprWithNewline(0) == snapshot


def test_trailing_newline_in_repr(snapshot):
    assert ReprWithNewline(1) == snapshot


def test_trailing_2_newlines_in_repr(snapshot):
    assert ReprWithNewline(2) == snapshot


class MultilineRepr:
    def __repr__(self) -> str:
        return "\n".join(
            [
                "Line1",
                "Line2\n",  # extra newline
                "Line3 ",  # with an extra space
            ]
        )


def test_multiline_repr(snapshot):
    assert MultilineRepr() == snapshot
