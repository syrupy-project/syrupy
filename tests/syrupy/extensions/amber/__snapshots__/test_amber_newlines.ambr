# serializer version: 1
# name: test_multiline_repr
  Line1
  Line2
  
  Line3 
# ---
# name: test_trailing_2_newlines_in_repr
  ReprWithNewline
  
  
# ---
# name: test_trailing_newline_in_repr
  ReprWithNewline
  
# ---
# name: test_trailing_no_newline_in_repr
  ReprWithNewline
# ---
